src/pulselistener/pulselistener/mercurial.py

```diff
@@ -80,20 +84,43 @@ def cleanup(self):
             assert isinstance(diff, dict)
             assert 'phid' in diff
 
+            failure = None
+            start = time.time()
             try:
-                await self.handle_diff(diff)
+                await self.handle_diff(build_target_phid, diff)
             except hglib.error.CommandError as e:
                 logger.warn('Mercurial error on diff', error=e.err, args=e.args, phid=diff['phid'])
 
-                # Remove uncommited changes
-                self.repo.revert(self.repo_dir.encode('utf-8'), all=True)
+                # Report mercurial failure as a Unit Test issue
+                failure = UnitResult(
+                    namespace='code-review',
+                    name='mercurial',
+                    result=UnitResultState.Fail,
+                    details='WARNING: The code review bot failed to apply your patch.\n\n```{}```'.format(e.err),
+                    format='remarkup',
+                    duration=time.time() - start,
+                )
 
             except Exception as e:
                 logger.warn('Failed to process diff', error=e, phid=diff['phid'])
 
+                # Report generic failure as a Unit Test issue
+                failure = UnitResult(
+                    namespace='code-review',
+                    name='mercurial',
+                    result=UnitResultState.Broken,
+                    details='WARNING: An error occured in the code review bot when applying your patch.\n\n```{}```'.format(e),
```

Comment:
This could be any error which happened during in the bot, right?
So maybe we shouldn't say "when applying your patch"

Reply:
And the name shouldn't be "mercurial".

Reply:
You're right... Any suggestion for the name ? It will be displayed as `code-review::XXX`. I suggest `general` or `workflow`

Reply:
`general` sounds good